### PR TITLE
Add PSTN calling handlers for hold, resume, and addParticipant

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Adds handlers to the calling component bindings and adapters for:
- holding a call
- resuming a call
- adding a participant
# Why

- starts to setup PSTN functionality in the repo

https://skype.visualstudio.com/SPOOL/_workitems/edit/2865531
# How Tested

Built to make sure stable API hasn't changed and ran samples to make sure all is well.

### CODE REVIEW

**Review by edwardlee-msft — APPROVED** [review]

File: common/scripts/babel-conditional-preprocess.js [hunk 1]
```diff
@@ -81,6 +81,10 @@ exports.default = babelHelper.declare((_api, opts) => {
         Handle(path, featureSet, stabilizedFeatureSet);
       },
 
+      TSMethodSignature(path) {
```
Comment (anjulgarg) [inline common/scripts/babel-conditional-preprocess.js:84]: @dmceachernmsft Can we do this as a separate PR?
With explanation to why we need to do this?
And possibly an example in the conditional compilation examples doc?

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md [hunk 2]
```diff
@@ -80,9 +82,12 @@ export type CallingHandlers = {
     onStopScreenShare: () => Promise<void>;
     onToggleScreenShare: () => Promise<void>;
     onHangUp: () => Promise<void>;
+    onHold: () => Promise<void>;
```
Comment (anjulgarg) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:85]: `onToggleHold` might be a better name since the same method will toggle between the hold states.

File: packages/calling-component-bindings/src/handlers/createHandlers.ts [hunk 3]
```diff
@@ -302,8 +318,31 @@ export const createDefaultCallingHandlers = memoizeOne(
...
+    const onAddParticipant = async (
+      participant:
+        | CommunicationUserIdentifier
+        | PhoneNumberIdentifier
+        | UnknownIdentifier
+        | MicrosoftTeamsUserIdentifier,
+      options?: AddPhoneNumberOptions
+    ): Promise<void> => {
+      const newACSorTeamsUser = participant as CommunicationUserIdentifier | MicrosoftTeamsUserIdentifier;
+      const newPSTNUser = participant as PhoneNumberIdentifier;
```
Comment (anjulgarg) [inline packages/calling-component-bindings/src/handlers/createHandlers.ts:331]: Wouldn't `if (newPSTNUser)` always be true?

Comment (dmceachernmsft) [reply packages/calling-component-bindings/src/handlers/createHandlers.ts:331]: if the casting to that type fails it will be undefined since the properties don't align for those types, so when passing in a participant only one of the casts will work.

File: packages/calling-component-bindings/src/handlers/createHandlers.ts [hunk 4]
```diff
@@ -302,8 +318,31 @@ export const createDefaultCallingHandlers = memoizeOne(
...
+      if (newPSTNUser) {
```
Comment (anjulgarg) [inline packages/calling-component-bindings/src/handlers/createHandlers.ts:332]: I would simplify all this logic by doing this:
```tsx
if (participant.phoneNumber) {
  await call?.addParticipant(participant as PhoneNumberIdentifier, options);    
} else {
  await call?.addParticipant(participant as CommunicationUserIdentifier | MicrosoftTeamsUserIdentifier);
}
```

File: packages/calling-component-bindings/src/handlers/createHandlers.ts (hunk 3 above)
Comment (anjulgarg) [reply packages/calling-component-bindings/src/handlers/createHandlers.ts:331]: The casting is compile time only and won't fail during run time.

File: packages/calling-component-bindings/src/handlers/createHandlers.ts (hunk 4 above)
Comment (anjulgarg) [reply packages/calling-component-bindings/src/handlers/createHandlers.ts:332]: OR use the `isPhoneNumberIdentifier` function exported by the communication-common package.
```tsx
import { isPhoneNumberIdentifier } from "@azure/communication-common";

if (isPhoneNumberIdentifier(participant)) {
  await call?.addParticipant(participant as PhoneNumberIdentifier, options);
} else {
  await call?.addParticipant(
    participant as CommunicationUserIdentifier | MicrosoftTeamsUserIdentifier
  );
}
```

File: common/scripts/babel-conditional-preprocess.js (hunk 1 above)
Comment (dmceachernmsft) [reply common/scripts/babel-conditional-preprocess.js:84]: @prprabhu-ms has a seperate PR for this as well I just included it so mine will build. its in the same spot so it wont conflict. (hopefully). https://github.com/Azure/communication-ui-library/pull/1913

File: packages/calling-component-bindings/src/handlers/createHandlers.ts (hunk 4 above)
Comment (anjulgarg) [reply packages/calling-component-bindings/src/handlers/createHandlers.ts:332]: Second option is better as it used the intended methods.

File: packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts [hunk 5]
```diff
@@ -415,6 +425,24 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
     this.handlers.onRemoveParticipant(userId);
   }
 
+  /* @conditional-compile-remove(PSTN-calls) */
+  public async addParticipant(
+    participant: PhoneNumberIdentifier | CommunicationUserIdentifier | UnknownIdentifier,
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts:430]: Should we have the `MicrosoftTeamsUserIdentifier` here as well?

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 6]
```diff
@@ -232,6 +240,29 @@ export interface CallAdapterCallManagement {
...
+  /* @conditional-compile-remove(PSTN-calls) */
+  /**
+   * Holds the call for the local user
+   * @beta
+   */
+  holdCall(): Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  /**
+   * resumes the call for the local user.
+   * @beta
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:263]: Nit: r -> R

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 7]
```diff
@@ -232,6 +240,29 @@ export interface CallAdapterCallManagement {
...
   removeParticipant(userId: string): Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  /**
+   * Adds a participant to the call by dialing them in
+   * @param participant - Phone number or Communication userId for participant being added
+   * @param options - Call options containing ACS number for PSTN calls
+   * @beta
+   */
+  addParticipant(
+    participant: PhoneNumberIdentifier | CommunicationUserIdentifier | UnknownIdentifier | MicrosoftTeamsUserIdentifier,
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:251]: Instead of creating a union of this type everywhere. Can we use the `CommunicationIdentifier` exported by communication-common?

File: packages/calling-component-bindings/src/handlers/createHandlers.ts (hunk 4 above)
Comment (dmceachernmsft) [reply packages/calling-component-bindings/src/handlers/createHandlers.ts:331]: oh cool! will update this

File: common/scripts/babel-conditional-preprocess.js (hunk 1 above)
Comment (prprabhu-ms) [reply common/scripts/babel-conditional-preprocess.js:84]: Merged mine. You can merge from `main` and get rid of this diff now.

Thanks for helping make conditional compilation exhaustive... slowly... incrementally.... some day. ;-)
